transfers/mover.go

```diff
@@ -0,0 +1,315 @@
+// Copyright (c) 2023 Cohere Consulting, LLC
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of
+// this software and associated documentation files (the "Software"), to deal in
+// the Software without restriction, including without limitation the rights to
+// use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+// of the Software, and to permit persons to whom the Software is furnished to do
+// so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package transfers
+
+import (
+	"encoding/gob"
+	"log/slog"
+	"path/filepath"
+	"time"
+
+	"github.com/google/uuid"
+
+	"github.com/kbase/dts/config"
+	"github.com/kbase/dts/endpoints"
+)
+
+//-------
+// Mover
+//-------
+
+// The mover manages actual file transfer operations and cancellations.
+
+// mover global state
+var mover moverState
+
+type moverState struct {
+	Channels  moverChannels
+	Endpoints map[string]endpoints.Endpoint
+}
+
+type moverChannels struct {
+	RequestMove         chan uuid.UUID
+	RequestCancellation chan uuid.UUID
+	Error               chan error
+	SaveAndStop         chan *gob.Encoder
+}
+
+func newMoverChannels() moverChannels {
+	return moverChannels{
+		RequestMove:         make(chan uuid.UUID, 32),
+		RequestCancellation: make(chan uuid.UUID, 32),
+		Error:               make(chan error, 32),
+		SaveAndStop:         make(chan *gob.Encoder),
+	}
+}
+
+func (channels *moverChannels) close() {
+	close(channels.RequestMove)
+	close(channels.RequestCancellation)
+	close(channels.Error)
+	close(channels.SaveAndStop)
+}
+
+// starts the mover
+func (m *moverState) Start() error {
+	slog.Debug("mover.Start")
+	m.Channels = newMoverChannels()
+	m.Endpoints = make(map[string]endpoints.Endpoint)
+	go m.process(nil)
+	return <-m.Channels.Error
+}
+
+// loads the mover from saved data
+func (m *moverState) Load(decoder *gob.Decoder) error {
+	slog.Debug("mover.Start")
+	m.Channels = newMoverChannels()
+	m.Endpoints = make(map[string]endpoints.Endpoint)
+	go m.process(decoder)
+	return <-m.Channels.Error
+}
+
+// stops the mover goroutine
+func (m *moverState) SaveAndStop(encoder *gob.Encoder) error {
+	slog.Debug("mover.Stop")
+	m.Channels.SaveAndStop <- encoder
+	err := <-m.Channels.Error
+	m.Channels.close()
+	return err
+}
+
+// starts moving files associated with the given transfer ID
```

Comment:
```suggestion
// starts moving files associated with the given transfer ID.
// a manifest generation will be triggered after a successful move
```
similar to comment about staging.